addons/download-button/userscript.js

```diff
@@ -0,0 +1,57 @@
+import downloadBlob from "../../libraries/common/cs/download-blob.js";
+
+export default async function ({ addon, console, msg }) {
+  const vm = addon.tab.traps.vm;
+  const { redux } = addon.tab;
+
+  await redux.waitForState((state) => typeof state.session.session?.user === "object");
+  const username = await addon.auth.fetchUsername();
+
+  await redux.waitForState((state) => state.preview.status.project === "FETCHED");
+  const projectAuthor = redux.state.preview.projectInfo.author?.username;
+
+  const isOwn = username === projectAuthor;
+  const shared = addon.tab.redux.state.preview.projectInfo.is_published;
+
+  async function download() {
+    const downloadButton = document.querySelector(".sa-download-button");
+    if (downloadButton.classList.contains("waiting")) return;
+    downloadButton.classList.add("loading");
+    try {
+      const project = await vm.saveProjectSb3();
+      const title = isOwn
+        ? await addon.tab.waitForElement(".project-title input")
+        : await addon.tab.waitForElement(".project-title");
```

Comment:
Correct me if I'm wrong but there elements are guaranteed to exist by this point, right?
```suggestion
        ? document.querySelector(".project-title input")
        : document.querySelector(".project-title");
```

Reply:
Uhm yeah they should be